public/app/core/components/NavBar/NavBarMenu.tsx

```diff
@@ -0,0 +1,112 @@
+import { CustomScrollbar, Icon, IconButton, IconName, useTheme2 } from '@grafana/ui';
+import { FocusScope } from '@react-aria/focus';
+import { useOverlay } from '@react-aria/overlays';
+import { css } from '@emotion/css';
+import { NavBarMenuItem } from './NavBarMenuItem';
+
+export interface Props {
+  activeItem?: NavModelItem;
+  navItems: NavModelItem[];
+  onClose: () => void;
+}
+
+export function NavBarMenu({ activeItem, navItems, onClose }: Props) {
+  const theme = useTheme2();
+  const styles = getStyles(theme);
+  const ref = useRef(null);
+  const { overlayProps } = useOverlay(
+    {
+      isDismissable: true,
+      isOpen: true,
+      onClose,
+    },
+    ref
+  );
+
+  return (
+    <FocusScope contain restoreFocus autoFocus>
+      <div data-testid="navbarmenu" className={styles.container} ref={ref} {...overlayProps}>
+        <div className={styles.header}>
+          <Icon name="bars" size="xl" />
+          <IconButton aria-label="Close navigation menu" name="times" onClick={onClose} size="xl" variant="secondary" />
+        </div>
+        <nav className={styles.content}>
+          <CustomScrollbar>
+            <ul>
+              {navItems.map((link, index) => (
+                <div className={styles.section} key={`${link.id}-${index}`}>
+                  <NavBarMenuItem
+                    isActive={activeItem === link}
+                    isSectionHeader
+                    onClick={() => {
+                      link.onClick?.();
+                      onClose();
+                    }}
+                    target={link.target}
+                    text={link.text}
+                    url={link.url}
+                  />
+                  {link.children?.map((childLink, childIndex) => (
+                    <NavBarMenuItem
+                      key={`${childLink.id}-${childIndex}`}
+                      icon={childLink.icon as IconName}
+                      isActive={activeItem === childLink}
+                      isDivider={childLink.divider}
+                      onClick={() => {
+                        childLink.onClick?.();
+                        onClose();
+                      }}
+                      target={childLink.target}
+                      text={childLink.text}
+                      url={childLink.url}
+                    />
+                  ))}
+                </div>
+              ))}
+            </ul>
+          </CustomScrollbar>
+        </nav>
+      </div>
+    </FocusScope>
+  );
+}
+
+NavBarMenu.displayName = 'NavBarMenu';
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  container: css`
+    background-color: ${theme.colors.background.canvas};
+    bottom: 0;
+    display: flex;
+    flex-direction: column;
+    left: 0;
+    min-width: 300px;
+    position: fixed;
+    right: 0;
+    top: 0;
+
+    ${theme.breakpoints.up('md')} {
+      border-right: 1px solid ${theme.colors.border.weak};
+      right: unset;
+    }
+  `,
```

Comment:
Do we prefer using CSS objects instead? 🤔

Reply:
@joshhunt sorry, don't quite follow what you mean by css objects here. do you mean pulling the `top`/`left`/`right`/`bottom` out of this component and instead putting it in the component that consumes it? 🤔

Reply:
Using

```ts
container: css({
  backgroundColor: theme.colors.background.canvas,
  ...
})
```

instead of 

```ts
  container: css`
    background-color: ${theme.colors.background.canvas};
    ...
`
```

I thought I heard somewhere that we prefer the former object style, instead of the tagged templates.

Reply:
ohhhhhhh, i had no idea you could even use that syntax. happy to change it to that if that's our preferred style 👍

Reply:
Yeah im not 100% sure on that one though. Personally I prefer it (less syntax noise, ts auto complete), and so when I think i heard that it is _our_ preferred style I really latched onto that 😅

Reply:
mmm, one to discuss at frontend weekly? i'm curious if there's any downsides. 🤔